src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -132,12 +147,76 @@ public void Initialize(TelemetryConfiguration configuration)
                             portalSender.Send(traceEvent);
                         }
 
+                        // set up heartbeat
+                        if (this.HeartbeatProvider == null)
+                        {
+                            this.HeartbeatProvider = new HealthHeartbeatProvider();
+                        }
+
+                        this.HeartbeatProvider.Initialize(configuration, this.TimeSpanBetweenHeartbeats, this.DisableHeartbeatProperties);
+
                         this.isInitialized = true;
                     }
                 }
             }
         }
-        
+
+        /// <summary>
+        /// Add a new Health Heartbeat property to the payload sent with each heartbeat.
+        /// 
+        /// To update the value of the property you are adding to the health heartbeat, 
+        /// <see cref="DiagnosticsTelemetryModule.SetHealthProperty"/>.
+        /// 
+        /// Note that you cannot add a HealthHeartbeatProperty with a name that already exists in the HealthHeartbeat
+        /// payload, including (but not limited to) the name of SDK-default items.
+        /// 
+        /// </summary>
+        /// <param name="payloadItem">Name, value, and health indicator of the value to add</param>
+        /// <returns>True if the new payload item is successfully added, false otherwise.</returns>
+        public bool AddHealthProperty(HealthHeartbeatProperty payloadItem)
```

Comment:
you can avoid adding entire class `HealthHeartbeatProperty` if there will be two methods - one to add properties and another to set health state. API can be as simple as:

``` csharp
IDictionaty<string, string> meta = module.HeartbeatMetadata;
meta.Add("aggregation module enabled", "true");

int state = module.HeartbeatHealthState;
module.HeartbeatHealthState += 1;
module.HeartbeatHealthState -=1;
```